File: shopping-list-et.api/shopping-list-et.domain/Entities/Item.cs
namespace shopping_list_et.domain.Entities
{
    public class Item
    {
        public int? Id { get; set; }
        public string? Text { get; set; }
        public DateTime? CreatedAt { get; set; }
        public bool? Checked { get; set; }
        
        public int ShoppingListId { get; set; }
        public ShoppingList? ShoppingList { get; set; }
    }
}

File: shopping-list-et.app/lib/models/item.dart
class Item{
  final int id;
  final String text;
  final DateTime createdAt;
  bool checked;

  Item(this.id, this.text, this.createdAt, this.checked);

  Item.fromJson(Map<String, dynamic> json)
      : id = json["id"],
        text = json["text"],
        createdAt = DateTime.parse(json["createdAt"]),
        checked = json["checked"];
}